examples/serve/authentication/server.py

```diff
@@ -0,0 +1,31 @@
+import fastapi
+from fastapi import security
+import uvicorn
+
+app = fastapi.FastAPI()
+
+bearer_security = security.HTTPBearer()
+STATIC_TOKEN = "static_secret_token"
+
+
+def verify_token(
+    credentials: security.HTTPAuthorizationCredentials = fastapi.Depends(
+        bearer_security),
+) -> None:
+    if credentials.credentials != STATIC_TOKEN:
+        raise fastapi.HTTPException(status_code=403,
+                                    detail="Invalid authentication credentials")
+
+
+@app.get("/health")
+async def health():
+    return {"status": "ok"}
+
+
+@app.get("/compute")
+async def compute(_=fastapi.Depends(verify_token)):
+    return {"message": "Token validation successful!"}
+
+
+if __name__ == "__main__":
+    uvicorn.run(app, host="0.0.0.0", port=8087)
```

Comment:
maybe can give an example of client program that accesses the server with the verify token

Reply:
+1

Reply:
We've already has one in the Task YAML's comment?

Reply:
Implement in python is rather challenging as I didn't find any default support for this; ref https://stackoverflow.com/questions/34359529/python-requests-equivalent-for-curls-location-trusted

Reply:
Hmm, not being able to programmatically interacting with the service seems like a key limitation in this mode.

Reply:
Yeah... I guess that is another reason we might want to use a proxy. Will investigate more

Reply:
Can we test it with OpenAI API? Maybe their library handles the authentication key automatically with redirection.

Reply:
I suppose the problem is our controller does not forward the request with previous auth headers..? Correct me if I'm wrong, but why does replica-side Openai API have a relation to this problem..?

Reply:
Oh, I mean the client side OpenAI python API, to see if the client querying the endpoint programmatically will have the similar behavior as `curl --trust-location`.

If this is an issue, we should start migrating our load balancer to proxy based solution.

Reply:
Just tried, the OpenAI Client does not support forwarding auth headers as well. It could access the replica endpoint but not the load balancer endpoint.

<img width="969" alt="image" src="https://github.com/skypilot-org/skypilot/assets/74357442/e5245c16-a08d-4ba6-89f3-c826be66fa3f">